src/main/java/org/pentaho/s3common/S3CommonPipedOutputStream.java

```diff
@@ -139,7 +139,7 @@ public Boolean call() throws Exception {
         int totalRead = 0;
         int partNum = 1;
         BufferedInputStream bis = new BufferedInputStream( pipedInputStream, PART_SIZE );
-        S3NWindowedSubstream s3is;
+        S3CommonWindowedSubstream s3is;
 
         while ( ( read = bis.read( tmpBuffer ) ) >= 0 ) {
```

Comment:
Could use try with resources for the ...Streams here. While it looks like "baos" is being closed appropriately, I don't see where "bis" is being closed.

Reply:
I tried to avoid making substantial changes in the output stream classes since they didn't need to be messed with much to fix the defect I was dealing with.